modules/core/src/test/java/org/apache/ignite/internal/causality/IncrementalVersionedValueTest.java

```diff
@@ -341,6 +344,51 @@ void testCompleteMultipleFutures() {
         assertThat(future3, willBe(2));
     }
 
+    /**
+     * Tests that {@link IncrementalVersionedValue#dependingOn(IncrementalVersionedValue)} provides causality between 2 different values.
+     */
+    @RepeatedTest(100)
+    public void testDependingOn() {
+        var vv0 = new IncrementalVersionedValue<>(register, () -> 1);
+
+        var vv1 = new IncrementalVersionedValue<>(dependingOn(vv0), () -> 1);
+
+        int token = 1;
+
+        vv0.update(token, (i, e) -> completedFuture(i + 1));
+
+        vv1.update(token, (i, e) -> completedFuture(i + 1));
+
+        register.moveRevision(token);
+
+        // Will always complete in time.
+        vv1.get(token).join();
```

Comment:
How about using `assertThat(..., willCompleteSuccessfully())` just in case there is a bug preventing the completion?

Reply:
Ok, why not. I just wanted to avoid as much CPU activity as I could for the next assertion